drivers/bluetooth/hci/cyw43xxx.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief CYW43xxx HCI extension driver.
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/uart.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cyw43xxx_driver);
+
+#include <stdint.h>
+
+#define HCI_UART_NODE                     DT_COMPAT_GET_ANY_STATUS_OKAY(infineon_cyw43xxx_bt_hci)
```

Comment:
I'd suggest declaring `DT_DRV_COMPAT` and using `DT_INST_...` macros instead (using instance 0). Saves from kicking the node id around. Then you could just do, for example `DT_INST_PROP(0, baudrate_for_fw_download)`